src/Engine/ProtoCore/Lang/FFIFunctionEndPoint.cs

```diff
@@ -97,10 +97,10 @@ public override StackValue Execute(Runtime.Context c, List<StackValue> formalPar
                 return StackValue.Null;
             }
 
-                // Check if this is a 'this' pointer function overload that was generated by the compiler
-            if (null != mFNode && mFNode.IsAutoGeneratedThisProc)
+            // Check if this is a 'this' pointer function overload that was generated by the compiler
+            const int thisPtrIndex = 0;
+            if (null != mFNode && mFNode.IsAutoGeneratedThisProc && formalParameters[thisPtrIndex].IsPointer)
```

Comment:
hmm @mmisol Can you explain why we do not need the `IsDefaultArgument` check below?

Reply:
I assume you mean the code that was there from line 119 to 122. That is no longer useful because we added `&& formalParameters[thisPtrIndex].IsPointer` to this if statement. The condition `formalParameters[thisPtrIndex].IsPointer || formalParameters[thisPtrIndex].IsDefaultArgument` will always be true inside this code block. Makes sense?

Reply:
not really 😉

Reply:
@mmisol This code block would be executed If "formalParameters[thisPtrIndex]" is a default argument but after your change, it will be executed only if "formalParameters[thisPtrIndex]" is a Pointer. Will this cause any change to the existing behavior?